app/Helpers/database/achievement.php

```diff
@@ -194,7 +194,8 @@ function UploadNewAchievement(
         return false;
     }
 
-    if ($type !== null && !AchievementType::isValid($type)) {
+    // TODO: remove "not-given" when RAIntegration supports `type`
```

Comment:
I'm not sure when that will occur. Even if a new version of RAIntegration were released tomorrow, we've been allowing versions all the way back to 1.0.4 because it's the last version that supported Windows XP, and there haven't been any critical anti-cheating changes released.

Reply:
I see. It sounds like then it may be best to always assume the potential presence of this fallback value and remove these TODO comments.

Reply:
Comments updated in https://github.com/RetroAchievements/RAWeb/pull/1783/commits/89db13bbf40d67f0122bb455f82a5159a0f9058a.